File: hw4/hw4/Calculator.cs
namespace hw1
{
    public static class Calculator
    {
        public static int Calculate(int val1, string operation, int val2)
        {
            var result = 0;
            result = operation switch
            {
                "+" => val1 + val2,
                "-" => val1 - val2,
                "*" => val1 * val2,
                "/" => val1 / val2,
                _ => result
            };
            return result;
        }
    }
}

File: hw6/hw5.FS/Calculator.fs
module hw5.Calculator
open hw5.ResBuilder
let Calculate (val1, operation, val2) =
    let tryDivide arg1 arg2 =
        if arg2 = decimal 0 then Error "Second argument is zero!"
        else Ok(arg1 / arg2)
    //help-method
    res{
        match operation with
        | Operation.Plus -> return val1 + val2
        | Operation.Minus -> return val1 - val2
        | Operation.Multiply -> return val1 * val2
        | _ -> let! res =  tryDivide val1 val2
               return res
        }